drivers/gpio/Kconfig.stm32

```diff
@@ -45,10 +45,9 @@ config GPIO_STM32_PORTJ
 config GPIO_STM32_PORTK
 	bool "Enable GPIO port K support"
 
-if SOC_SERIES_STM32F1X
```

Comment:
This change is abusive.
There are 5 symbols under the if statement.

Reply:
The `if` is around a `choice`, so it's fine. These two turn into the same thing after parsing:

```
if FOO

choice
	prompt "choice"

...

endchoice

endif
```

```
choice
	prompt "choice"
	depends on FOO

...

endchoice
```

All `if`s are removed and turned into `depends on` by the code.

Reply:
My bad I did not see all the symbols were part of the choice.
This is fine then